src/SFA.DAS.Apim.Developer.Application/AzureApimManagement/Services/UserService.cs

```diff
@@ -140,27 +149,50 @@ private async Task<UserResponse> UpsertApimUser(string apimUserId, UserDetails u
             }
             
             return apimUserResponse.Body;
-           
         }
 
         public async Task<UserDetails> CheckUserAuthentication(string email, string password)
         {
             var authenticatedTask =
                 _azureUserAuthenticationManagementService.GetAuthentication<GetUserAuthenticatedResponse>(
                     new GetUserAuthenticatedRequest(email, password));
-
             var userTask = GetUser(email);
-
             await Task.WhenAll(authenticatedTask, userTask);
 
             if (userTask.Result == null)
             {
                 return null;
             }
+            var user = userTask.Result;
             
-            userTask.Result.Authenticated = authenticatedTask.Result.StatusCode == HttpStatusCode.OK;
-            
-            return userTask.Result;
+            if (user.Note.ThirdFailedAuthDateTime.HasValue && DateTime.Now.AddMinutes(-10) > user.Note.ThirdFailedAuthDateTime)
+            {
+                user.Note.FailedAuthCount = 0;
+                user.Note.ThirdFailedAuthDateTime = null;
+                user.State = "active";
+                await UpsertApimUser(user.Id, user);
+                return await CheckUserAuthentication(email, password);
+            }
+
+            user.Authenticated = authenticatedTask.Result.StatusCode == HttpStatusCode.OK;
+            if (!user.Authenticated && !user.Note.ThirdFailedAuthDateTime.HasValue)
+            {
+                user.Note.FailedAuthCount += 1;
+                if (user.Note.FailedAuthCount >= 3)
+                {
+                    user.Note.ThirdFailedAuthDateTime = DateTime.Now;
+                    user.State = "blocked";
+                }
+                await UpsertApimUser(user.Id, user);
+            }
+
+            if (user.Authenticated && user.Note.FailedAuthCount > 0)
+            {
+                user.Note.FailedAuthCount = 0;
+                await UpsertApimUser(user.Id, user);
+            }
+
```

Comment:
Is it possible to pull out the `3` for FailedAuthCount into a constant in configuration? Should we ever need to increase or decrease the value, and with that change `ThirdFailedAuthDateTime` to `AccountLockedDateTime` ?